frame/sudo/src/extensions/check_only_sudo.rs

```diff
@@ -0,0 +1,106 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{Config, Pallet};
+use codec::{Decode, Encode};
+use frame_support::dispatch::DispatchInfo;
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{DispatchInfoOf, Dispatchable, SignedExtension},
+	transaction_validity::{
+		InvalidTransaction, TransactionPriority, TransactionValidity, TransactionValidityError,
+		ValidTransaction,
+	},
+};
+use sp_std::{fmt, marker::PhantomData};
+
+/// Ensure that signed transactions are only valid if they are signed by root.
+#[derive(Clone, Eq, PartialEq, Encode, Decode, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct CheckOnlySudo<T: Config + Send + Sync>(PhantomData<T>);
```

Comment:
```suggestion
pub struct AllowOnlySudoAccount<T: Config + Send + Sync>(PhantomData<T>);
```

The Metadata name should also be renamed to match this.

Reply:
I wanna keep similar naming with the extension of frame-system and I think `CheckOnlySudo` is clear enough.

Reply:
The name is really not descriptive at all. This is the reason I proposed a better name.

Reply:
Rename `CheckOnlySudo` ==> `CheckOnlySudoAccount`